tests/helpers/test_cherrypicker_data_helper.py

```diff
@@ -0,0 +1,320 @@
+
+        # Assert that expected fields contain the correct prefix and barcode
+        assert all([row[0].startswith(f"RSID-{barcode}") for row in barcode_rows])
+        assert all([row[1].startswith(f"VPID-{barcode}") for row in barcode_rows])
+        assert all([row[2].startswith(f"{barcode}_") for row in barcode_rows])
+        assert all([row[3].startswith(f"RNA_PCR-{barcode}") for row in barcode_rows])
+
+        # Check the correct number of positives and negatives were generated
+        positive_rows = [row for row in barcode_rows if "Positive" == row[4]]
+        negative_rows = [row for row in barcode_rows if "Negative" == row[4]]
+        assert len(positive_rows) == positives
+        assert len(negative_rows) == 96 - positives
+
+
+@pytest.fixture
+def test_rows_data():
+    return [
+        ["RSID-01", "VPID-01", "RNAID-01", "RNAPCRID-01", "Positive", "Timestamp", "AP"] + [""] * 12,
+        ["RSID-02", "VPID-02", "RNAID-02", "RNAPCRID-02", "Negative", "Timestamp", "AP"] + [""] * 12,
+    ]
+
+
+@pytest.fixture
+def expected_test_output():
+    return """Root Sample ID,Viral Prep ID,RNA ID,RNA-PCR ID,Result,Date Tested,Lab ID,CH1-Target,CH1-Result,CH1-Cq,CH2-Target,CH2-Result,CH2-Cq,CH3-Target,CH3-Result,CH3-Cq,CH4-Target,CH4-Result,CH4-Cq
+RSID-01,VPID-01,RNAID-01,RNAPCRID-01,Positive,Timestamp,AP,,,,,,,,,,,,
+RSID-02,VPID-02,RNAID-02,RNAPCRID-02,Negative,Timestamp,AP,,,,,,,,,,,,
+"""  # noqa E501
+
+
+@pytest.mark.parametrize("existing_output_path", [True, False])
+def test_write_plates_file_success(existing_output_path, test_rows_data, expected_test_output, logger_messages):
+    data_path = os.path.join("tmp", "data")
+    output_path = os.path.join(data_path, "TEST")
+    filename = "testing.csv"
+
+    shutil.rmtree(data_path, ignore_errors=True)
+
+    try:
+        if existing_output_path:
+            os.makedirs(output_path)
+
+        write_plates_file(test_rows_data, output_path, filename)
+
+        with open(os.path.join(output_path, filename), mode="r") as f:
+            saved_data = f.read()
+        assert saved_data == expected_test_output
+
+    finally:
+        shutil.rmtree(data_path, ignore_errors=True)
+
+    assert len(logger_messages.error) == 0
+    assert is_found_in_list("Writing to file", logger_messages.info)
+    assert is_found_in_list("Test data plates file written", logger_messages.info)
+    assert is_found_in_list("testing.csv", logger_messages.info)
+
+
+def test_write_plates_file_exception(test_rows_data, logger_messages):
+    output_path = os.path.join("tmp", "data", "TEST")
+    filename = "testing.csv"
+
+    with patch("builtins.open", side_effect=OSError(5, "Unable to write file")):
+        with pytest.raises(OSError):
+            write_plates_file(test_rows_data, output_path, filename)
+
+    assert is_found_in_list("Exception", logger_messages.error)
+    assert is_found_in_list("Unable to write file", logger_messages.error)
+
+
+def test_create_barcode_meta():
```

Comment:
To clarify, the idea is to create the label printing list in mongo on the crawler side linked to the run identifier, so it can be pulled out on the lighthouse side for label printing?

Reply:
Yes that's what's happening.  This method creates the metadata and then it gets added it to the Mongo document in the `jobs` module.  Lighthouse will see that the run is marked as completed and can then pick that up from the document to be able to send the print off to sprint.